File: tripin-service/src/main/java/grepp/NBE5_6_2_Team03/domain/user/mail/VerificationCodeProvider.java
package grepp.NBE5_6_2_Team03.domain.user.mail;

import grepp.NBE5_6_2_Team03.global.exception.BusinessException;
import org.springframework.stereotype.Component;

import java.security.SecureRandom;

@Component
public class VerificationCodeProvider implements CodeProvidable {
    private static final SecureRandom RANDOM = new SecureRandom();
    private static final int CODE_LENGTH = 10;

    @Override
    public boolean isSupport(CodeType codeType) {
        return codeType.isVerificationCode();
    }

    @Override
    public String provide(CodeType codeType) {
        if(codeType.isNotVerificationCode()){
            throw new BusinessException("CODE_TYPE 예외");
        }

        StringBuilder sb = new StringBuilder(CODE_LENGTH);
        for (int i = 0; i < CODE_LENGTH; i++) {
            int randomNum = RANDOM.nextInt(CODE_LENGTH);
            sb.append(randomNum);
        }
        return sb.toString();
    }
}

File: tripin-mail/src/main/kotlin/com/grepp/mailservice/service/strategy/VerificationCodeProvider.kt
package com.grepp.mailservice.service.strategy

import org.springframework.stereotype.Component
import java.security.SecureRandom

@Component
class VerificationCodeProvider : CodeProvidable {

    companion object {
        private val RANDOM = SecureRandom()
        private const val CODE_LENGTH = 10
    }

    override fun isSupport(codeType: CodeType): Boolean = codeType == CodeType.VERIFICATION_CODE
    override fun provide(): String = (1..CODE_LENGTH)
        .map { RANDOM.nextInt(10) }
        .joinToString("")
}
